cpp/src/arrow/util/io_util_test.cc

```diff
@@ -446,6 +446,56 @@ TEST(CreateDirTree, Basics) {
   ASSERT_OK_AND_ASSIGN(fn, temp_dir->path().Join("EF"));
   ASSERT_OK_AND_ASSIGN(created, CreateDirTree(fn));
   ASSERT_TRUE(created);
+
+#ifndef __APPLE__
```

Comment:
Why did you have to disable this test on macOS?

Reply:
I experienced failures in the CI pipeline on macOS and I was unable to locate clear documentation of the path name limits (I'm not a macOS expert). 

I figured it might be best to separately address macOS in another issue, if there's community support for that.  (This patch leaves macOS builds and tests unchanged.)

If you believe this test should run as is under macOS, I'll remove the #ifndef and follow up on any issues.

Reply:
I was just wondering. According to Google searches, the path length limit on macOS may be 1024, which this test exceeds. We can keep it like that.